### PR TITLE
feat: import device with template,quickly migrate device

### PR DESCRIPTION
**Pull request by rikugun**
import device with template,quickly migrate device

1. open device config
2. click import device from template
QQ_1730184923284
3. select devices.json from another fuxa export
QQ_1730184973111
4. modify device name
QQ_1730185031099

### CODE REVIEW

**Comment by rikugun** [comment]
import device will generate new device id and  tag id

File: client/src/app/device/device.component.html [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <mat-icon>more_vert</mat-icon>
 </button>
 <input type="file" #fileImportInput style="display: none" id="devicesConfigFileUpload" (change)="onFileChangeListener($event)" accept=".json,.csv" />
```
Comment (unocelli) [inline client/src/app/device/device.component.html:13]: change onFileChangeListener($event) to onDevTplChangeListener($event)

File: client/src/app/device/device.component.html [hunk 2]
```diff
@@ -11,6 +11,7 @@
...
+<input type="file" #tplFileImportInput style="display: none" id="devicesConfigTplUpload" (change)="onDevTplChangeListener($event)" accept=".json,.csv" />
```
Comment (unocelli) [inline client/src/app/device/device.component.html:14]: add flag parameter onDevTplChangeListener($event, true)

File: client/src/app/device/device.component.ts [hunk 3]
```diff
@@ -174,4 +180,43 @@ export class DeviceComponent implements OnInit, OnDestroy {
         reader.readAsText(input.files[0]);
         this.fileImportInput.nativeElement.value = null;
     }
+
+    onDevTplChangeListener(event){
```
Comment (unocelli) [inline client/src/app/device/device.component.ts:184]: add flag parameter  onDevTplChangeListener(event, isTemplate: boolean)

File: client/src/app/device/device.component.ts [hunk 4]
```diff
@@ -174,4 +180,43 @@ export class DeviceComponent implements OnInit, OnDestroy {
...
+                    device.name = Utils.getShortGUID(device.name + '_', '');
+                    if (device.tags) {
+                        Object.keys(device.tags).forEach((key) => {
+                            device.tags[key].id = Utils.getGUID(TAG_PREFIX);
+                        });
+                    }
+                    importDev.push(device);
+                }
+            });
+            this.projectService.importDevices(importDev);
```
Comment (unocelli) [inline client/src/app/device/device.component.ts:210]: this.projectService.importDevices(isTemplate ? importDev : devices);

**Comment by unocelli** [comment]
Good job, Thanks a lot!

**Pull request merged**